core/include/detray/coordinates/cartesian2.hpp

```diff
@@ -17,7 +17,7 @@ namespace detray {
 
 /** Frame projection into a cartesian coordinate frame
  */
-template <typename transform3_t>
+template <typename transform3_t = __plugin::transform3<detray::scalar>>
```

Comment:
I thought we are not supposed to use `__plugin::transform3`  anymore

Reply:
I don't really have an alternative right now, can we leave it for this PR?

Reply:
Can you explain what is the problem?

Reply:
Currently, I do template specialization on complete coordinate types. I think it can be changed to specializations on ```template<template>``` types, but that is more verbose and I would prefer to do that in a later PR, because I have work that is based on this PR

Reply:
ok

Reply:
Specializing on template template parameters somehow did not work, so I went with the very verbose solution